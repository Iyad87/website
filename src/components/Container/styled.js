import styled from 'styled-components';

import { gridWidth, gridWidthWide, gridColWidth } from '../../styles/spacing';

const maxWidth = {
  normal: `${gridWidth + gridColWidth}px`,
  wide: `${gridWidthWide + gridColWidth}px`,
};

export const StyledContainer = styled.div`
  margin: 0 auto;
  padding: 0 ${gridColWidth / 2}px;
  max-width: ${props => (props.wide ? maxWidth.wide : maxWidth.normal)};
`;
