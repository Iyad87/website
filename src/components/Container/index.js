import styled from 'styled-components';

import { gridWidth, gridColWidth, spacing } from '../../styles/spacing';

const Container = styled.div`
  margin: 0 auto;
  padding: 0 ${gridColWidth}px;
  max-width: ${gridWidth + gridColWidth}px;
`;

export default Container;
