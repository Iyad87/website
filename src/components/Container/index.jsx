import styled from 'styled-components';

import { gridWidth, gridWidthWide, gridColWidth } from '../../styles/spacing';

const maxWidth = {
  normal: `${gridWidth + gridColWidth}px`,
  wide: `${gridWidthWide + gridColWidth}px`,
};

const Container = styled.div`
  margin: 0 auto;
  padding: 0 ${gridColWidth / 2}px;
  max-width: ${props => (props.wide ? maxWidth.wide : maxWidth.normal)};
`;

export default Container;
