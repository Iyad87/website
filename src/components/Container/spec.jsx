import React from 'react';
import { shallow } from 'enzyme';
import Container from './index';

test('components/Container snapshot', () => {
  const wrapper = shallow(<Container>Some content</Container>);

  expect(wrapper).toMatchSnapshot();
});

test('components/Container snapshot (wide)', () => {
  const wrapper = shallow(<Container wide>Some content</Container>);

  expect(wrapper).toMatchSnapshot();
});
