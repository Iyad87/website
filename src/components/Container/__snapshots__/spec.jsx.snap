// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Container snapshot (wide) 1`] = `
<div
  className="sc-bdVaJa cAUiFA"
>
  Some content
</div>
`;

exports[`components/Container snapshot 1`] = `
<div
  className="sc-bdVaJa ipwjvv"
>
  Some content
</div>
`;
